src/schemas/base-types.yaml

```
@@ -47,6 +47,10 @@ uint256:
   title: hex encoded 256 bit unsigned integer
   type: string
   pattern: ^0x([1-9a-f]+[0-9a-f]{0,31})|0$
+int:
+  title: decimal signed integer
+  type: string
+  pattern: ^-?[0-9]+$
```

This does not match `uint` definition above it. either both should be decimal or both hex encoded. I understand that this will decrees readability but it has to be done for the sake of consistency.
```suggestion
int:
  title: hex encoded signed integer
  type: string
  pattern: ^0x([1-9a-f]+[0-9a-f]*|0)$
```

---

This is only used for error codes, consistent with how other errors are returned (but in general don't have proper error responses/types defined). As an example, eth_call can return:
```
{
  "jsonrpc": "2.0",
  "id": 1,
  "error": {
    "code": -32000,
    "message": "missing trie node 3a3d6059576515928999d764cbf49a2358ba38af3f399b6363d33f85e07ec5ce (path ) <nil>"
  }
}
``` 
but eth_call's definition gives no indication about error response/format and it does return a negative decimal
https://github.com/ethereum/execution-apis/blob/4da6a02f094d9d8c826272f6ceac2d751866f132/src/eth/execute.yaml#L1

---

I understand, but since it is "base-type", shouldn't follow the same format as other types around it. What if future API calls needed this type outside of the error codes scope? I think defining this should follow convention, and if you want to define another type for decimal int then you probably should. also in this case, type should be number and not string. because afaik `string` type means there is quotations around it. and in the code example shown, it does not have a quotation around it.
 maybe it should be something like this:
```suggestion
errorCode:
  title: decimal signed integer
  type: number
```

---

You are correct about it being incorrectly specified as string, that should be changed (and tested thoroughly, only arrays and and strings have ever been defined before).

We could rename to errorCode, although nothing else in that file is prescriptive about how that data type is being used, it's low-level types and it's up to the actual definition of the api to say:
```
    errorCode:
      title: Error Code
      $ref: '#/components/schemas/int'
```

---

number have been defined before:
you can see similar approach here:
https://github.com/ethereum/execution-apis/blob/fa0580e5f6ba89a9bbe71279bd99e6afce9b02fc/src/eth/fee_market.yaml#L38-L40

---

Keep in mind this is only a spec. What matters is the client implementations to be tested and make sure that they match the spec. the more descriptive the spec the better for all clients. One issue with number is it allows not only integers but also numbers with decimal point (float for example) so there the desciption should make sure to describe this well to client implementers.

---

Number has been used before, as applied to a field, but not as a base-type (and i'm not sure it would be clear it is signed, and should not be float). Ideally, this would be defined as an array of possible, explicit values, as here, and just drop the entire notion of signed int, although this has only been done for newer endpoints, it seems:
https://github.com/ethereum/execution-apis/blob/fa0580e5f6ba89a9bbe71279bd99e6afce9b02fc/src/engine/openrpc/methods/forkchoice.yaml#L19

---

Agree with the above approach. lets drop this int base type then and go with the above ^^^

---

This is even better hence it produces a specific list of numbers and defines what each error code means explicitly. much more prefered

---

Can we mark this as resolved now?  I think we agree that there should not be an `int` type, and instead we should just use literals for specific errors?